src/ui.jl

```diff
@@ -47,39 +47,42 @@ end
 TerminalMenus.cancel(m::CthulhuMenu) = m.selected = -1
 
-function stringify(@nospecialize(f), io::IO=IOBuffer())
-    f(IOContext(io, :color=>true))
-    return String(take!(io))
+stringify(@nospecialize(f), io::IO = devnull) = stringify(f, IOContext(io, :color=>true))
+function stringify(@nospecialize(f), context::IOContext)
+    buf = IOBuffer()
+    io = IOContext(buf, context)
+    f(io)
+    return String(take!(buf))
 end
 
 const debugcolors = (:nothing, :light_black, :yellow)
 function usage(@nospecialize(view_cmd), optimize, iswarn, hide_type_stable, debuginfo, remarks, with_effects, inline_cost, type_annotations, highlight)
-    colorize(iotmp, use_color::Bool, c::Char) = stringify(iotmp) do io
+    colorize(use_color::Bool, c::Char) = stringify() do io
         use_color ? printstyled(io, c; color=:cyan) : print(io, c)
     end
 
-    io, iotmp = IOBuffer(), IOBuffer()
+    io = IOBuffer()
     ioctx = IOContext(io, :color=>true)
 
     println(ioctx, "Select a call to descend into or ↩ to ascend. [q]uit. [b]ookmark.")
     println(ioctx, "Toggles: [",
-        colorize(iotmp, optimize, 'o'), "]ptimize, [",
-        colorize(iotmp, iswarn, 'w'), "]arn, [",
-        colorize(iotmp, hide_type_stable, 'h'), "]ide type-stable statements, [",
-        stringify(iotmp) do io
+        colorize(optimize, 'o'), "]ptimize, [",
+        colorize(iswarn, 'w'), "]arn, [",
+        colorize(hide_type_stable, 'h'), "]ide type-stable statements, [",
+        stringify() do io
             printstyled(io, 'd'; color=debugcolors[Int(debuginfo)+1])
         end, "]ebuginfo, [",
-        colorize(iotmp, remarks, 'r'), "]emarks, [",
-        colorize(iotmp, with_effects, 'e'), "]ffects, [",
-        colorize(iotmp, inline_cost, 'i'), "]nlining costs, [",
-        colorize(iotmp, type_annotations, 't'), "]ype annotations, [",
-        colorize(iotmp, highlight, 's'), "]yntax highlight for Source/LLVM/Native.")
+        colorize(remarks, 'r'), "]emarks, [",
+        colorize(with_effects, 'e'), "]ffects, [",
+        colorize(inline_cost, 'i'), "]nlining costs, [",
+        colorize(type_annotations, 't'), "]ype annotations, [",
+        colorize(highlight, 's'), "]yntax highlight for Source/LLVM/Native.")
     println(ioctx, "Show: [",
-        colorize(iotmp, view_cmd === cthulhu_source, 'S'), "]ource code, [",
-        colorize(iotmp, view_cmd === cthulhu_ast, 'A'), "]ST, [",
-        colorize(iotmp, view_cmd === cthulhu_typed, 'T'), "]yped code, [",
-        colorize(iotmp, view_cmd === cthulhu_llvm, 'L'), "]LVM IR, [",
-        colorize(iotmp, view_cmd === cthulhu_native, 'N'), "]ative code")
+        colorize(view_cmd === cthulhu_source, 'S'), "]ource code, [",
+        colorize(view_cmd === cthulhu_ast, 'A'), "]ST, [",
+        colorize(view_cmd === cthulhu_typed, 'T'), "]yped code, [",
+        colorize(view_cmd === cthulhu_llvm, 'L'), "]LVM IR, [",
+        colorize(view_cmd === cthulhu_native, 'N'), "]ative code")
```

Comment:
Won't this just always print to stdout? I don't think we want that

Reply:
Nah, now `stringify` function has been refactored so that it always creates a new `IOBuffer` object to generate a string from a given printing function

Reply:
Ah, ok! I was just confused since I thought `colorize` actually printed to the io instead of just outputting a string.